File: mini-projet/src/main/java/tn/esprit/spring/entities/CategorieProduit.java
package tn.esprit.spring.entities;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CategorieProduit implements Serializable{
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long idCategorie;
	private String libelle;
	
	@OneToMany(mappedBy = "categorieProduit", cascade = CascadeType.REMOVE)
	List<Produit> produits = new ArrayList<>();

}

File: miniprojet4/src/app/Model/CategorieProduit.ts
import {Produit} from "./Produit";

export class CategorieProduit {
  idCategorieProduit: number;
  libelle: string;
  categorieProduitIcone:string;
  produits:Produit[];


  constructor(idCategorieProduit: number, libelle: string, categorieProduitIcone: string) {
    this.idCategorieProduit = idCategorieProduit;
    this.libelle = libelle;
    this.categorieProduitIcone = categorieProduitIcone;
  }
}
